File: web/js/add.js
window.onload = function() {
    document.getElementById("ID").focus();
};
// function validateForm() {
//     var id = document.forms["dataForm"]["id"];
//     var number = document.forms["dataForm"]["number"];
//     var name = document.forms["dataForm"]["name"];
//
//     var idRegex = /^[a-zA-Z0-9_-]{3,16}$/;
//     var numberRegex = /^[0-9]{1,10}$/;
//     var nameRegex = /^[a-zA-Z\s]+$/;
//
//     if (!idRegex.test(id.value)) {
//         alert("Please enter a valid ID (3-16 characters, letters, numbers, - or _)");
//         return false;
//     }
//
//     if (!numberRegex.test(number.value)) {
//         alert("Please enter a valid Number (1-10 digits)");
//         return false;
//     }
//
//     if (!nameRegex.test(name.value)) {
//         alert("Please enter a valid Name (letters and spaces only)");
//         return false;
//     }
// }

function calculateDollar() {
    var rmb = parseFloat(document.getElementById("rmb").value);
    var dollar = rmb / 7.25;
    document.getElementById("dollar").value = dollar.toFixed(2);
}

// function replaceBlank() {
// // 获取表单
//     var form = document.getElementById("dataForm");
//
// // 遍历表单中的所有元素
//     for (var i = 0; i < form.elements.length; i++) {
//         var element = form.elements[i];
//
//         // 根据元素的名称属性判断是否需要处理
//         switch(element.name) {
//             case "id":
//             case "number":
//             case "name":
//             case "cDate":
//             case "belOrganization":
//             case "contactPerson":
//             case "phone":
//             case "organization":
//             case "nation":
//             case "delegate":
//             case "contactMan":
//             case "inSpectOrg":
//             case "inspector":
//                 // 使用正则表达式替换所有空格
//                 var trimmedValue = element.value.replace(/\s/g, '');
//                 // 将去除空格后的值重新赋给输入框
//                 element.value = trimmedValue;
//                 break;
//             // 如果是数字类型的属性，则不需要去除空格
//             case "rmb":
//             case "dollar":
//             case "years":
//                 break;
//             // 如果是日期类型的属性，需要根据实际情况处理，这里假设不需要去除空格
//             case "saleDate":
//             case "dateInspect":
//                 break;
//             // 如果是布尔类型的属性，这里假设不需要处理
//             case "conclusion":
//                 break;
//             // 默认情况下，不做处理
//             default:
//                 break;
//         }
//     }
//
// }

function confirmSubmit() {
    // 弹出确认提示框
    const confirmed = confirm("确定要更改吗？");

    // 如果用户点击了确认按钮，则返回 true 继续提交表单；否则返回 false 阻止表单提交
    return confirmed;
}

File: src/com/cc/servlet/Add.java
package com.cc.servlet;

import com.cc.domain.Sales;
import com.cc.service.SalesService;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class Add extends HttpServlet {
    public void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        request.setCharacterEncoding("UTF-8");
        response.setContentType("text/html;charset=utf-8");

        String id = request.getParameter("id");
        String number = request.getParameter("number");
        String name = request.getParameter("name");
        String startDateString = request.getParameter("startDate");
        String endDateString = request.getParameter("endDate");
        String belOrganization = request.getParameter("belOrganization");
        String contactPerson = request.getParameter("contactPerson");
        String phone = request.getParameter("phone");
        double rmb = Double.parseDouble(request.getParameter("rmb"));
        double dollar = Double.parseDouble(request.getParameter("dollar"));
        int years = Integer.parseInt(request.getParameter("years"));
        String saleDateString = request.getParameter("saleDate");
        String organization = request.getParameter("organization");
        String nation = request.getParameter("nation");
        String delegate = request.getParameter("delegate");
        String contactMan = request.getParameter("contactMan");
        String inSpectOrg = request.getParameter("inSpectOrg");
        boolean conclusion = request.getParameter("conclusion") != null;
        String inspector = request.getParameter("inspector");
        String dateInspectString = request.getParameter("dateInspect");

        // 将日期字符串转换为 Date 对象
        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
        SimpleDateFormat toFormat = new SimpleDateFormat("yyyy年MM月dd日");
        Date startDate = null;
        Date endDate = null;
        Date saleDate = null;
        Date dateInspect = null;
        try {
            startDate = dateFormat.parse(startDateString);
            endDate = dateFormat.parse(endDateString);
            startDateString = toFormat.format(startDate);
            endDateString = toFormat.format(endDate);
            saleDate = dateFormat.parse(saleDateString);
            dateInspect = dateFormat.parse(dateInspectString);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        String cDate = startDateString+"-"+endDateString;

        Sales sales = new Sales(
                id,
                number,
                name,
                cDate,
                belOrganization,
                contactPerson,
                phone,
                rmb,
                dollar,
                years,
                saleDate,
                organization,
                nation,
                delegate,
                contactMan,
                inSpectOrg,
                conclusion,
                inspector,
                dateInspect
        );
        request.getSession().setAttribute("adding", sales);
        SalesService ss = new SalesService();
        if (!ss.getByID(id).isEmpty()){
           response.getWriter().println("<script language='javascript'>alert('id已存在');window.location.href='/Wed/jsp/add.jsp';</script>");
        } else if(ss.addSale(sales)){
            request.getSession().setAttribute("adding", "");
            response.getWriter().println("<script language='javascript'>confirm('添加成功');window.location.href='/Wed/jsp/add.jsp';</script>");
//            response.sendRedirect("/Wed/jsp/add.jsp");
        }else {
           response.getWriter().println("<script language='javascript'>alert('添加失败');window.location.href='/Wed/jsp/add.jsp';</script>");
        }
    }

    public void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        this.doGet(request, response);
    }

}
